assets/scripts/modules/Modal.js

```diff
@@ -0,0 +1,195 @@
+            },
+
+            onPostActivate: () => {
+                this.$togglers.forEach(($toggler) => {
+                    $toggler.setAttribute('aria-expanded', true)
+                })
+            },
+
+            onDeactivate: () => {
+                this.el.classList.remove(Modal.CLASS.EL)
+                $html.classList.remove(Modal.CLASS.HTML)
+                $html.classList.remove('has-'+this.dataName+'-open')
+                this.el.setAttribute('aria-hidden', true)
+                this.isOpen = false
+
+                this.onDeactivate?.();
+            },
+
+            onPostDeactivate: () => {
+                this.$togglers.forEach(($toggler) => {
+                    $toggler.setAttribute('aria-expanded', false)
+                })
+            },
+
+            clickOutsideDeactivates: true,
+        }
+
+        this.isOpen = false
+    }
+
+    /////////////////
+    // Lifecycle
+    /////////////////
+    init() {
+        this.onBeforeInit?.()
+
+        this.focusTrap = createFocusTrap(
+            this.$focusTrapTargets.length > 0 ? this.$focusTrapTargets : [this.el],
+            this.focusTrapOptions
+        )
+
+        this.bindEvents()
+
+        this.onInit?.()
+    }
+
+    destroy() {
+        this.focusTrap?.deactivate?.({
+            returnFocus: false,
+        })
+
+        this.unbindEvents()
+
+        this.onDestroy?.()
+
+        super.destroy()
+    }
+
+    /////////////////
+    // Events
+    /////////////////
+    bindEvents() {
+        window.addEventListener(CUSTOM_EVENT.VISIT_START, this.onVisitStart)
+        window.addEventListener(CUSTOM_EVENT.MODAL_OPEN, this.onModalOpen)
+
+        this.$togglers.forEach(($toggler) => {
+            $toggler.addEventListener('click', this.toggle)
+        })
+    }
+
+    unbindEvents() {
+        window.removeEventListener(CUSTOM_EVENT.VISIT_START, this.onVisitStart)
+        window.removeEventListener(CUSTOM_EVENT.MODAL_OPEN, this.onModalOpen)
+
+        this.$togglers.forEach(($toggler) => {
+            $toggler.removeEventListener('click', this.toggle)
+        })
+    }
+
+    /////////////////
+    // Callbacks
+    /////////////////
+    onVisitStart() {
+        // Close the modal on page change
+        this.close()
+    }
+
+    onModalOpen(event) {
+        // Close the modal if another one is opened
+        if (event.detail !== this.el) {
+            this.close()
+        }
+    }
+
+    /////////////////
+    // Methods
+    /////////////////
+    toggle(event) {
+        if (this.el.classList.contains(Modal.CLASS.EL)) {
+            this.close(event)
+        } else {
+            this.open(event)
+        }
+    }
+
+    open(args) {
+        if (this.isOpen) return
+
+        this.focusTrap?.activate?.()
+
+        this.onOpen?.(args)
+
+        window.dispatchEvent(new CustomEvent(CUSTOM_EVENT.MODAL_OPEN, { detail: this.el }))
+    }
+
+    close(args) {
+        if (!this.isOpen) return
```

Comment:
```suggestion
        if (!this.isOpen) {
            return
        }
```

The compiler is going to minify this anyway, might as well maximize readability.